Kernel/Firmware/EFI/Protocols/ConsoleSupport.h

```diff
@@ -0,0 +1,182 @@
+    };
+
+    enum class BackgroundColor : i32 {
+        Black = 0x00,
+        Blue = 0x01,
+        Green = 0x02,
+        Cyan = 0x03,
+        Red = 0x04,
+        Magenta = 0x05,
+        Brown = 0x06,
+        LightGray = 0x07,
+    };
+
+    ForegroundColor foreground_color : 4;
+    BackgroundColor background_color : 3;
+    i32 : 32 - (4 + 3);
+};
+static_assert(AssertSize<TextAttribute, 4>());
+
+// SIMPLE_TEXT_OUTPUT_MODE: See "Related Definitions" at https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#simple-text-output-protocol
+struct SimpleTextOutputMode {
+    i32 max_mode;
+    i32 mode;
+    TextAttribute attribute;
+    i32 cursor_column;
+    i32 cursor_row;
+    Boolean cursor_visible;
+};
+static_assert(AssertSize<SimpleTextOutputMode, 24>());
+
+// EFI_SIMPLE_TEXT_OUTPUT_PROTOCOL: https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#simple-text-output-protocol
+struct SimpleTextOutputProtocol {
+    static constexpr GUID guid = { 0x387477c2, 0x69c7, 0x11d2, { 0x8e, 0x39, 0x00, 0xa0, 0xc9, 0x69, 0x72, 0x3b } };
+
+    using TextResetFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, Boolean extended_verification);
+    using TextStringFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, char16_t* string);
+    using TextTestStringFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, char16_t* string);
+    using TextQueryModeFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, FlatPtr mode_number, FlatPtr* columns, FlatPtr* rows);
+    using TextSetModeFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, FlatPtr mode_number);
+    using TextSetAttributeFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, TextAttribute attribute);
+    using TextClearScreenFn = EFIAPI Status (*)(SimpleTextOutputProtocol*);
+    using TextSetCursorPositionFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, FlatPtr column, FlatPtr row);
+    using TextEnableCursorFn = EFIAPI Status (*)(SimpleTextOutputProtocol*, Boolean visible);
+
+    TextResetFn reset;
+    TextStringFn output_string;
+    TextTestStringFn test_string;
+    TextQueryModeFn query_mode;
+    TextSetModeFn set_mode;
+    TextSetAttributeFn set_attribute;
+    TextClearScreenFn clear_screen;
+    TextSetCursorPositionFn set_cursor_position;
+    TextEnableCursorFn enable_cursor;
+    SimpleTextOutputMode* mode;
+};
+static_assert(AssertSize<SimpleTextOutputProtocol, 80>());
+static_assert(__builtin_offsetof(SimpleTextOutputProtocol, output_string) == 8);
+
+// EFI_GRAPHICS_OUTPUT_BLT_PIXEL: https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#efi-graphics-output-protocol-blt
+struct GraphicsOutputBltPixel {
+    u8 blue;
+    u8 green;
+    u8 red;
+    u8 reserved;
+};
+
+// EFI_GRAPHICS_OUTPUT_BLT_OPERATION: https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#efi-graphics-output-protocol-blt
+enum class GraphicsOutputBltOperation {
+    VideoFill,
+    VideoToBltBuffer,
+    BufferToVideo,
+    VideoToVideo,
+    Max
```

Comment:
nit: trailing comma on last element in enum

(I understand there won't be more entries after it in this case; it's still a nice blanket style rule)